src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -2030,15 +2050,19 @@ UpdateShardMoveDependencies(PlacementUpdateEvent *move, uint64 colocationId, int
 		shardMoveDependencies.colocationDependencies, &colocationId, HASH_ENTER, NULL);
 	shardMoveDependencyInfo->taskId = taskId;
 
-	shardMoveDependencyInfo = hash_search(shardMoveDependencies.nodeDependencies,
-										  &move->sourceNode->nodeId, HASH_ENTER, NULL);
-
-	shardMoveDependencyInfo->taskId = taskId;
-
-	shardMoveDependencyInfo = hash_search(shardMoveDependencies.nodeDependencies,
-										  &move->targetNode->nodeId, HASH_ENTER, NULL);
-
-	shardMoveDependencyInfo->taskId = taskId;
+	bool found;
+	ShardMoveSourceNodeHashEntry *shardMoveSourceNodeHashEntry = hash_search(
+		shardMoveDependencies.nodeDependencies, &move->sourceNode->nodeId, HASH_ENTER,
+		&found);
+	if (found)
+	{
+		shardMoveSourceNodeHashEntry->taskIds = lappend_int(
```

Comment:
taskId is of type int64.

Also, can we have duplicate dependencies here? Are duplicates eliminated before inserting to pg_dist_background_task_depend ?

Consider using a HTAB for shardMoveSourceNodeHashEntry->taskIds.

Reply:
> Also, can we have duplicate dependencies here? Are duplicates eliminated before inserting to pg_dist_background_task_depend ?

Duplicates are handled in `GenerateTaskMoveDependencyList` through `HTAB dependsList`

> taskId is of type int64
Right, thanks for pointing it out. I didn't know that int lists store int32